content/docs/platform/subscription/headless-hooks.mdx

```diff
@@ -0,0 +1,157 @@
+           <span>{subscription.name || subscription.identifier}</span>
+        </li>
+      ))}
+    </ul>
+  );
+}
+```
+
+## Get a single subscription
+
+If you need to manage a specific subscription instance, use the `useSubscription` hook. It fetches the details and preference states for a single subscription.
+
+```tsx
+import { useSubscription } from '@novu/react';
+
+function ProjectSettings() {
+  const { subscription, isLoading } = useSubscription({ 
+    topicKey: 'project-123',
+    identifier: 'user-project-alert'
+  });
+
+  if (isLoading) return <div>Loading...</div>;
+
+  if (!subscription) return <div>You are not subscribed to this project.</div>;
+
+  return (
+    <div>
+      <h3>Subscription: {subscription.identifier}</h3>
+      <pre>{JSON.stringify(subscription.preferences, null, 2)}</pre>
+    </div>
+  );
+}
+```
+
+## Create a subscription
+
+To allow users to opt-in to a topic, use the `useCreateSubscription` hook. This is often used on "Follow" or "Subscribe" buttons.
+
+```tsx
+import { useCreateSubscription } from '@novu/react';
+
+function SubscribeButton() {
+  const { create, isCreating } = useCreateSubscription();
+
+  const handleSubscribe = async () => {
+    try {
+      await create({ 
+        topicKey: 'project-123', 
+        identifier: 'user-project-alert',
+        // Optional: Set initial preferences
+        preferences: [] 
+      });
+      console.log('Subscribed successfully!');
+    } catch (error) {
+      console.error('Failed to subscribe');
+    }
+  };
+
+  return (
+    <button onClick={handleSubscribe} disabled={isCreating}>
+      {isCreating ? 'Subscribing...' : 'Subscribe to Updates'}
+    </button>
+  );
+}
+```
+
+## Update a subscription
+
+Use `useUpdateSubscription` to modify an existing subscription. This is primarily used to toggle workflow preferences, enabling, or disabling specific notifications within the subscription.
+
+```tsx
+import { useUpdateSubscription } from '@novu/react';
+
+function PreferenceToggle({ subscriptionId, workflowId, isEnabled }) {
+  const { update, isUpdating } = useUpdateSubscription();
+
+  const togglePreference = async () => {
+    await update({
+      topicKey: 'project-123',
+      subscriptionId: subscriptionId,
+      preferences: [
+        { 
+          workflowId: workflowId, 
+          enabled: !isEnabled // Toggle the boolean value
+        }
+      ]
+    });
+  };
+
+  return (
+    <input 
+      type="checkbox" 
+      checked={isEnabled} 
+      onChange={togglePreference}
+      disabled={isUpdating} 
+    />
+  );
+}
+```
+
+## Delete a subscription
+
+To allow users to opt-out or unsubscribe, use the `useDeleteSubscription` hook.
+
+```tsx
+import { useDeleteSubscription } from '@novu/react';
+
+function UnsubscribeButton({ subscriptionId }) {
+  // Alias 'delete' to 'deleteSubscription' to avoid syntax errors
+  const { delete: deleteSubscription, isDeleting } = useDeleteSubscription();
```

Comment:
if we have a syntax error with the `delete` prop, I will consider renaming this hook or prop; I will update you on that